brian2/tests/test_neurongroup.py

```diff
@@ -654,7 +655,7 @@ def test_threshold_reset():
                     threshold='v > 1', reset='v=0.5')
     G.v = np.array([0, 1, 2])
     run(defaultclock.dt)
-    assert_equal(G.v[:], np.array([0, 1, 0.5]))
+    assert_allclose(G.v[:], np.array([0, 1, 0.5]))
```

Comment:
Comment not about this particular instance, but I wonder if we lose anything by weakening our tests like this? My feeling is probably not, but it makes me mildly anxious. I guess you've already pretty much covered this with the float-type dependent version of allclose.

Reply:
In general I think we need to do a better job separating our tests for basic functionality from the tests for mathematical/numerical correctness. Currently, we only have a few tests (e.g. for STDP) that compare the exact results for non-trivial simulations.

Reply:
Worth creating an issue for? I'm not entirely sure it's necessary, I feel like ``allclose`` is probably as good as ``equal`` practically. Can we think of an example where it might be a problem that it passes allclose but not equal?